ldi-orchestrator/ldio-connectors/ldio-archive-file-in/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/LdioArchiveFileInAutoConfig.java

```diff
@@ -0,0 +1,52 @@
+package be.vlaanderen.informatievlaanderen.ldes.ldio.config;
+
+import be.vlaanderen.informatievlaanderen.ldes.ldi.services.ComponentExecutor;
+import be.vlaanderen.informatievlaanderen.ldes.ldi.types.LdiAdapter;
+import be.vlaanderen.informatievlaanderen.ldes.ldi.types.LdiComponent;
+import be.vlaanderen.informatievlaanderen.ldes.ldio.ArchiveFileCrawler;
+import be.vlaanderen.informatievlaanderen.ldes.ldio.LdioArchiveFileIn;
+import be.vlaanderen.informatievlaanderen.ldes.ldio.LdioArchiveFileInRunner;
+import be.vlaanderen.informatievlaanderen.ldes.ldio.configurator.LdioInputConfigurator;
+import be.vlaanderen.informatievlaanderen.ldes.ldio.valueobjects.ComponentProperties;
+import org.apache.jena.riot.Lang;
+import org.apache.jena.riot.RDFLanguages;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+@Configuration
+public class LdioArchiveFileInAutoConfig {
+
+	public static final String ARCHIVE_ROOT_DIR_PROP = "archive-root-dir";
+	public static final String SOURCE_FORMAT = "source-format";
+
+	@Bean("be.vlaanderen.informatievlaanderen.ldes.ldio.LdioArchiveFileIn")
+	public LdioInputConfigurator ldiArchiveFileInConfigurator() {
+		return new LdioInputConfigurator() {
+			@Override
+			public LdiComponent configure(LdiAdapter adapter,
+					ComponentExecutor executor,
+					ComponentProperties config) {
+				ArchiveFileCrawler archiveFileCrawler = new ArchiveFileCrawler(getArchiveDirectoryPath(config));
+				Lang sourceFormat = getSourceFormat(config);
```

Comment:
I follow this reasoning when leaving the class, but here everything stays in the scope of the LdioInputConfigurator class. In this case I prefered to isolate the logic on how to handle every property in a separate method. Maybe it would be better to have a separate class where I inject the config as a final field but currently that would feel like overengineering.